plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/expression/ExpressionConverter.java

```diff
@@ -0,0 +1,260 @@
+        }
+        // Null checks
+        else if (StandardFunctions.IS_NULL_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            finalHudiExpression = Predicates.isNull(arguments.getFirst());
+        }
+        // IN operator from Trino's perspective might be a call like "IN"(column, val1, val2...)
+        // or "IN"(column, ARRAY[val1, val2...]) or it might be handled by TupleDomain primarily.
+        // If it comes as a function:
+        else if (StandardFunctions.IN_PREDICATE_FUNCTION_NAME.equals(functionName) && arguments.size() > 1) {
+            // First argument is the column/expression, rest are values or an array
+            finalHudiExpression = Predicates.in(arguments.getFirst(), arguments.subList(1, arguments.size()));
+        }
+        // Cast
+        else if (StandardFunctions.CAST_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            Expression sourceExpression = arguments.getFirst();
+            finalHudiExpression = new CastExpression(sourceExpression, hudiReturnType);
+        }
+        // Functions
+        else if (HudiTrinoDayFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            return new HudiTrinoDayFunctionExpression(functionName.getName(), arguments, hudiReturnType);
+        }
+        else if (HudiTrinoFromUnixFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            // The number of arguments will determine which overload it is.
+            // Trino's type system will provide the correct trinoReturnType based on the overload.
+            // (e.g., TIMESTAMP for 1-arg, TIMESTAMP WITH TIME ZONE for 2/3-arg versions)
+            // No need to force a correct returnType here; expression should handle it.
+            finalHudiExpression = new HudiTrinoFromUnixFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoYearFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoYearFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoMonthFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoMonthFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoHourFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
+            checkArgumentSize(1, functionName.getName(), arguments);
+            finalHudiExpression = new HudiTrinoHourFunctionExpression(arguments, hudiReturnType);
+        }
+        else if (HudiTrinoDateFormatFunctionExpression.TRINO_FN_NAME.equalsIgnoreCase(functionName.getName())) {
```

Comment:
For the specific functions for expression index, should they be put into a separate method for handling so it's more clear?

Reply:
Was thinking of this, but not really sure how to do it as initialisation of the concrete implementation is currently a one liner. It is concise and readable and i can't think of any better way to make it more readable.